azurerm/resource_arm_application_insights_webtests_test.go

```diff
@@ -0,0 +1,275 @@
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+					resource.TestCheckResourceAttr(resourceName, "geo_locations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "frequency", "300"),
+					resource.TestCheckResourceAttr(resourceName, "timeout", "30"),
+				),
+			},
+			{
+				Config: testAccAzureRMApplicationInsightsWebTests_complete(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+					resource.TestCheckResourceAttr(resourceName, "geo_locations.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "frequency", "900"),
+					resource.TestCheckResourceAttr(resourceName, "timeout", "120"),
+				),
+			},
+			{
+				Config: testAccAzureRMApplicationInsightsWebTests_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+					resource.TestCheckResourceAttr(resourceName, "geo_locations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "frequency", "300"),
+					resource.TestCheckResourceAttr(resourceName, "timeout", "30"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApplicationInsightsWebTests_requiresImport(t *testing.T) {
+	if !requireResourcesToBeImported {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_application_insights_webtest.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMApplicationInsightsWebTests_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsWebTestsDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMApplicationInsightsWebTests_requiresImport(ri, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_application_insights_webtest"),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMApplicationInsightsWebTestsDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).appInsightsWebTestsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_application_insights_webtest" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		id, err := parseAzureResourceID(rs.Primary.Attributes["id"])
+		if err != nil {
+			return err
+		}
+		resGroup := id.ResourceGroup
```

Comment:
rather than doing this we can do:

```suggestion
		resGroup := rs.Primary.Attributes["resource_group_name"]
```